src/libraries/System.Private.CoreLib/src/System/Numerics/Vector2.cs

```diff
@@ -326,6 +326,26 @@ public static float Dot(Vector2 value1, Vector2 value2)
                  + (value1.Y * value2.Y);
         }
 
+        /// <summary>
+        /// Returns the z-value of the cross product of two vectors.
+        /// Since the Vector2 is in the x-y plane, a 3D cross product only produces the z-value.
+        /// </summary>
+        /// <param name="value1">The first vector.</param>
+        /// <param name="value2">The second vector.</param>
+        /// <returns>The value of the z-coordinate from the cross product.</returns>
+        /// <remarks>
+        /// Return z-value = value1.X * value2.Y - value1.Y * value2.X
+        /// <see cref="Cross"/> is the same as taking the <see cref="Dot"/> with the second vector
+        /// that has been rotated 90-degrees.
+        /// </remarks>
+        [Intrinsic]
```

Comment:
my understanding is that as long as you don't contribute a JIT change, then it won't get replaced by JIT with more optimal codegen

@tannergooding please confirm

```suggestion
```